lib/internal/modules/cjs/loader.js

```diff
@@ -854,9 +854,14 @@ Module._load = function(request, parent, isMain) {
 };
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (StringPrototypeStartsWith(request, 'node:') ||
-      (NativeModule.canBeRequiredByUsers(request) &&
-      NativeModule.canBeRequiredWithoutScheme(request))) {
+  if ((
+    StringPrototypeStartsWith(request, 'node:') &&
+    NativeModule.canBeRequiredByUsers(StringPrototypeSlice(request, 5))
+  ) ||
+  (
+    NativeModule.canBeRequiredByUsers(request) &&
+    NativeModule.canBeRequiredWithoutScheme(request)
+  )) {
```

Comment:
nit: I find it easier to read like this, feel free to ignore this suggestion if you disagree.
```suggestion
  if (
    (
      StringPrototypeStartsWith(request, 'node:') &&
      NativeModule.canBeRequiredByUsers(StringPrototypeSlice(request, 5))
    ) || (
      NativeModule.canBeRequiredByUsers(request) &&
      NativeModule.canBeRequiredWithoutScheme(request)
    )
  ) {
```